ActivityLog: display number of activities selected when `showAppliedFiltersCount` prop is passed

Initsogar opened the pull request:
Resolves https://github.com/Automattic/jetpack-backup-team/issues/203

## Proposed Changes
- Display number of activities selected instead of the name of each activity, by using an optional `showAppliedFiltersCount` prop.
  - This will apply only when there's more than 1 checkbox selected.
- Add a new `showAppliedFiltersCount` prop to `TypeSelector`
- Move render logic for selector button to its function
- Rename `Activity Type` to `Activity type`

| Scenario | Changes description | Before | After |
|---|---|---|---|
| Default state | The button is now labeled as `Activity type` | ![CleanShot 2023-11-02 at 08 56 29](https://github.com/Automattic/wp-calypso/assets/1488641/dad5f9a0-0326-49c9-90c8-f836077c88a3) | ![CleanShot 2023-11-02 at 08 56 01](https://github.com/Automattic/wp-calypso/assets/1488641/954cde1c-cf20-4872-9493-5d1d5ac7aa81) |
| 1 selected | No changes | ![CleanShot 2023-11-02 at 08 53 23](https://github.com/Automattic/wp-calypso/assets/1488641/eb65db65-48e3-4464-82a7-736a0181cabc) | ![CleanShot 2023-11-02 at 08 53 23](https://github.com/Automattic/wp-calypso/assets/1488641/63c53380-4c74-426e-bace-0e706cbf4338) |
| More than 1 selected | The count of activities selected are now displayed instead of the name of each activity. | ![CleanShot 2023-11-02 at 08 59 17](https://github.com/Automattic/wp-calypso/assets/1488641/a9690798-262f-498f-b5a4-39f51aa8b691) | ![CleanShot 2023-11-02 at 08 58 05](https://github.com/Automattic/wp-calypso/assets/1488641/98b2f37a-3e85-4fb6-8107-9db84bb366c5) |

## Demo
### Calypso Blue
![CleanShot 2023-11-02 at 09 04 43](https://github.com/Automattic/wp-calypso/assets/1488641/90277023-5d1c-46d4-b9a0-8eddc9a66d85)

### Jetpack Cloud
![CleanShot 2023-11-02 at 09 07 25](https://github.com/Automattic/wp-calypso/assets/1488641/9e66e97f-6e32-4bfc-9001-82506857f941)

## Testing Instructions
* Spin up a Calypso and Jetpack Cloud live branch
* Navigate to the Activity Log on each
* Ensure that selecting `Activity Type` filters makes the button displays `X activities selected` instead of the name of each filter. Similar behavior described in the screenshots shared above.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Initsogar committed:
Display number of types selected when necessary

- Add a new showAppliedFiltersCount prop to `TypeSelector`
- Move render logic for selector button to its own function

Initsogar commented on client/my-sites/activity/filterbar/type-selector/activity-type-selector.jsx:
```diff
@@ -12,7 +12,13 @@ import { TypeSelector } from './type-selector';
+		<TypeSelector
+			{ ...props }
+			title={ translate( 'Activity Type' ) }
+			showAppliedFiltersCount={ true }
```
`ActivityTypeSelector` is only used by the Activity Log, so we need to default it here as `true`, given that `TypeSelector` is also [used by `IssueTypeSelector` on Jetpack Manage](https://github.com/Automattic/wp-calypso//blob/16453d9100df016ac8e3d1c1d5799d6f5b8b01ad/client/my-sites/activity/filterbar/type-selector/issue-type-selector.tsx#L13).

Initsogar committed:
Add translations to activities selected text

keoshi reviewed (commented):
Hey @Initsogar, this is looking really good.

When testing this out I found that the labels are still taking a lot of space because of the label I suggested `X activities selected`. If possible, I'd propose a different logic that also keeps real labels for longer:

- **No value selected**: show the default state (nothing changes).
- **1-2 values selected**: show the actual labels for selected activities (changes).
- **2+ values selected**: show just the number of activities (changes).

Here's how that looks like:

![image](https://github.com/Automattic/wp-calypso/assets/390760/30da758e-ad83-4554-9ad1-82c41ce943c5)

What do you think? Worth changing?

---

Unrelated, but the biggest issue I see right now is that the dropdown selector moves around when the selected label grows. This will be solved when the filters are left aligned, but the solution above could also help prevent huge jumps.

![Nov-02-2023 08-21-23](https://github.com/Automattic/wp-calypso/assets/390760/3bac025c-74cd-4c2f-921b-2a8e577e765a)

Initsogar committed:
Replace activity type title to sentence case

Initsogar committed:
Display filters count when more than 1 is selected

Initsogar commented:
@keoshi thanks for the feedback!

Based on our recent discussion I just push new changes:
- Renamed `Activity Type` to `Activity type`
- Added the count of filters selected only when more than 1 activity is selected
- Replaced `X activities selected` with `X selected`

Also updated the description of this PR with the new proposed changes.

Could you take another look?

> Unrelated, but the biggest issue I see right now is that the dropdown selector moves around when the selected label grows. This will be solved when the filters are left aligned, but the solution above could also help prevent huge jumps.

It seems this is still happening if you first check `Backups and Restores`, but given we are switching the label to `X selected` after clicking more than one, there will be no more big jumps.

keoshi reviewed (approved):
Looks great to me!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9583859

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Initsogar for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9583859).
